lib/Serialization/DeserializeSIL.cpp

```diff
@@ -1538,16 +1557,21 @@ bool SILDeserializer::readSILInstruction(SILFunction *Fn, SILBasicBlock *BB,
   case SILInstructionKind::StringLiteralInst: {
     Identifier StringVal = MF->getIdentifier(ValID);
     auto encoding = fromStableStringEncoding(Attr);
-    if (!encoding) return true;
+    if (!encoding) {
+      MF->error();
```

Comment:
Should MF->error() be `[[unreachable]]` or whatever?

Reply:
:-/ It's stateful at the moment—if you error early on in loading the module, it records that information for use later and compilation can keep going. I should probably go rework that at some point so that the recoverable parts go through a different entry point.